contracts/NFTVesting/IntervalVestingNFTTimeLock.sol

```diff
@@ -0,0 +1,169 @@
+     * `beneficiary_` when {release} is invoked after `releaseTime_`. The release time is specified as a Unix timestamp
+     * (in seconds).
+     */
+    constructor(
+        IERC721 nft_,
+        uint256 tokenId_,
+        address beneficiary_,
+        uint256 releaseTime_,
+        uint256 maxDiscount_,
+        uint256 maxInterval_,
+        uint256 intervalDuration_
+    ) {
+        require(
+            releaseTime_ > block.timestamp,
+            "TimeLock: release time is before current time"
+        );
+        _nft = nft_;
+        _tokenId = tokenId_;
+        _beneficiary = beneficiary_;
+        _releaseTime = releaseTime_;
+        _maxDiscount = maxDiscount_;
+        _maxInterval = maxInterval_;
+        _intervalDuration = intervalDuration_;
+        _currentInterval = 0;
+    }
+
+    /**
+     * @dev Returns the smart contract NFT.
+     */
+    function nft() public view virtual returns (IERC721) {
+        return _nft;
+    }
+
+    /**
+     * @dev Returns the token ID of the NFT being held.
+     */
+    function tokenId() public view virtual returns (uint256) {
+        return _tokenId;
+    }
+
+    /**
+     * @dev Returns the beneficiary that will receive the NFT.
+     */
+    function beneficiary() public view virtual returns (address) {
+        return _beneficiary;
+    }
+
+    /**
+     * @dev Returns the time when the NFT are released in seconds since Unix Interval (i.e. Unix timestamp).
+     */
+    function releaseTime() public view virtual returns (uint256) {
+        return _releaseTime;
+    }
+
+    /**
+     * @dev Returns discount for locking at each Interval
+     */
+    function discountPerInterval() public view virtual returns (uint256) {
+        return _maxDiscount / _maxInterval;
+    }
+
+    /**
+     * @dev Returns current Interval
+     */
+    function getCurrentInterval() public view returns (uint256) {
+        if (block.timestamp < _releaseTime) {
+            return 0;
+        }
+        uint256 interval = uint256(block.timestamp - _releaseTime) /
+            _intervalDuration;
+        if (interval > _maxInterval) {
+            interval = _maxInterval;
+        }
+        return interval;
+    }
+
+    /**
+     * @dev Returns the remaining interval before max vesting interval
+     */
+    function getIntervalsLeft() public view returns (uint256) {
+        return _maxInterval - _currentInterval;
+    }
+
+    /**
+     * @dev Updates the interval after release schedule
+     */
+    function updateInterval() private validRelease {
+        require(_currentInterval <= _maxInterval, "Vesting Schedule is over.");
+        // floor the current Interval to the nearest Interval interval
+        _currentInterval = getCurrentInterval();
+    }
+
+    /**
+     * @dev Get the current discount in terms of percentage for the vesting schedule.
+     */
+    function getDiscount() public view returns (uint256) {
+        require(_currentInterval > 0, "Vesting Schedule is not Up yet.");
```

Comment:
This seems to connotate that unless >= 1 interval has passed, the NFT cannot be unlocked which is different from all the other smart contracts in which one just gets less discount - intended?

Reply:
yup i was thinking before the first interval, the locker cannot unlock and therefore would not have any discount. For the other contracts, there is a cliff time which is the `releaseTime`

Reply:
Gotcha then this needs to be part of the documentation - i.e. that there is a minimum lock time